### PR TITLE
Engine integration for sysmon linux

### LINKED ISSUE #17396: Engine - Sysmon (linux) integration
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
|5.0.0 | Engine | Manager | Packages/Sources | OS version |

This issue is to make an enhancement to the Wazuh engine to expand its ruleset and improve the overall user experience. Specifically, the integration of sysmon (linux) into the engine's list of supported integrations.

To ensure compatibility and interoperability, this sysmon (linux) integration should adhere to the Elastic Common Schema (ECS) in all its fields. By following ECS guidelines, we can maintain consistency and facilitate seamless data management across different systems.

As part of this integration, we must track and report the number of fields successfully extracted from sysmon (linux), as well as those that could not be extracted, along with the reasons for their exclusion. This information will assist users in understanding the data availability and any potential limitations they may encounter.

We aim to achieve the following objectives:

1. Expanded Ruleset: Increase the coverage of rules within the Wazuh engine, enabling more comprehensive security monitoring and threat detection capabilities.
2. Improved User Experience: Provide users with a seamless and user-friendly experience when utilizing Wazuh's sysmon (linux) integration, making it easier to extract and analyze relevant data.
3. ECS Compliance: Ensure that all fields within the sysmon (linux) integration align with the Elastic Common Schema, promoting interoperability and simplifying data management.

## Task List:
- [x] Obtain a list and a pre-analysis of the possible event types with the default configuration.
- [x] Create a decoder to extract the useful fields of each event type.
- [x] Analysis of fields that can be obtained but there is no clear destination in the ECS.
- [x] Analysis of fields that cannot be extracted (parse). 
- [x] Generate end to end events and check filbeat logs to rule out errors
- [x] Add test and expected for all event types

### PR DESCRIPTION
**Pull request by juliancnn**
|Related issue|
|---|
| Closes #17396 |

The sysmon integration is an extension of the sysmen integration because by default, the configuration writes to the same `/var/log/syslog` file.  The logs are syslog type, where the message is an XML that we will process with the windows decoder parser, that will transform the Arrays data in objects with Key that will be the attribute name of the Object array.

Currently, sysmon for linux version 1.2.0 can detect the following events:

| Event ID | Description                                                       |
| -------- | ----------------------------------------------------------------- |
| 1        | Logs when a new process is created.                               |
| 3        | Logs TCP/UDP connections on the machine.                          |
| 4        | Logs the state of the Sysmon service (started or stopped).        |
| 5        | Logs when a process terminates.                                   |
| 9        | Logs when a process conducts reading operations, from the drive.  |
| 11       | Logs when a file is created or overwritten.                       |
| 16       | Logs when the local Sysmon configuration is updated.              |
| 23       | Logs when a file is deleted by a process.                         |

Within the events not all fields are implemented, for example in this version no hashing algorithm is implemented.

ref:
- [Sysmon event doc](https://learn.microsoft.com/en-us/sysinternals/downloads/sysmon)
- [Linux supported event](https://github.com/OTRF/OSSEM-DD/tree/main/linux/sysmon)
- [Hashes](https://github.com/Sysinternals/SysmonForLinux/issues/131)

### Sysmon config:
```xml
<Sysmon schemaversion="4.81">
  <HashAlgorithms>*</HashAlgorithms>
	<EventFiltering>
        	<!-- Event ID 1 == ProcessCreate. Log all newly created processes -->
		<ProcessCreate onmatch="exclude"/>
		 <!-- Event ID 3 == NetworkConnect Detected. Log all network connections -->
		<NetworkConnect onmatch="exclude"/>
		<!-- Event ID 4 == Sysmon service state changed, non filter -->
		<!-- Event ID 5 == ProcessTerminate. Log all processes terminated -->
		<ProcessTerminate onmatch="exclude" />
		<!-- Event ID 9 == RawAccessRead. Log all raw access read -->
		<RawAccessRead onmatch="exclude"/>
		<!-- Event ID 10 == ProcessAccess. Log all open process operations -->
		<ProcessAccess onmatch="exclude"/>
        <!-- Event ID 11 == FileCreate. Log every file creation -->
        <FileCreate onmatch="exclude" />
    	<!-- Event ID 16 == ServiceConfigurationChange, non filter -->
        <!--Event ID 23 == FileDelete. Log all files being deleted -->
		<FileDelete onmatch="exclude" />
	</EventFiltering>
</Sysmon>

```

### Sample events

```xml
<!--
  Event id 16 == sysmon config change
  Event id  4 == sysmon service start
  systemctl restart sysmon
-->
Jun  5 18:52:50 202-u20-dev-agent sysmon:<Event>
  <System>
    <Provider Name="Linux-Sysmon" Guid="{ff032593-a8d3-4f13-b0d6-01fc615a0f97}"/>
    <EventID>16</EventID>
    <Version>3</Version>
    <Level>4</Level>
    <Task>16</Task>
    <Opcode>0</Opcode>
    <Keywords>0x8000000000000000</Keywords>
    <TimeCreated SystemTime="2023-06-05T18:52:50.923175000Z"/>
    <EventRecordID>944</EventRecordID>
    <Correlation/>
    <Execution ProcessID="4365" ThreadID="4365"/>
    <Channel>Linux-Sysmon/Operational</Channel>
    <Computer>202-u20-dev-agent</Computer>
    <Security UserId="0"/>
  </System>
  <EventData>
    <Data Name="UtcTime">2023-06-05 18:52:50.923</Data>
    <Data Name="Configuration">sysconfig.xml</Data>
    <Data Name="ConfigurationFileHash">-</Data>
  </EventData>
</Event>
Jun  5 18:52:50 202-u20-dev-agent sysmon: <Event>
<System>
  <Provider Name="Linux-Sysmon" Guid="{ff032593-a8d3-4f13-b0d6-01fc615a0f97}"/>
  <EventID>4</EventID>
  <Version>3</Version>
  <Level>4</Level>
  <Task>4</Task>
  <Opcode>0</Opcode>
  <Keywords>0x8000000000000000</Keywords>
  <TimeCreated SystemTime="2023-06-05T18:52:50.923275000Z"/>
  <EventRecordID>945</EventRecordID>
  <Correlation/>
  <Execution ProcessID="4365" ThreadID="4365"/>
  <Channel>Linux-Sysmon/Operational</Channel>
  <Computer>202-u20-dev-agent</Computer>
  <Security UserId="0"/>
</System>
<EventData>
  <Data Name="UtcTime">2023-06-05 18:52:50.923</Data>
  <Data Name="State">Started</Data>
  <Data Name="Version">1.2.0</Data>
  <Data Name="SchemaVersion">4.81</Data>
</EventData>
</Event>


<!--
  Event id 1 ==  ProcessCreate
  Event id 11 ==  FileCreate
  Event id 5 ==  ProcessTerminate
  touch /tmp/e11.log
-->
Jun  5 18:50:35 202-u20-dev-agent sysmon:<Event>
<System>
  <Provider Name="Linux-Sysmon" Guid="{ff032593-a8d3-4f13-b0d6-01fc615a0f97}"/>
  <EventID>1</EventID>
  <Version>5</Version>
  <Level>4</Level>
  <Task>1</Task>
  <Opcode>0</Opcode>
  <Keywords>0x8000000000000000</Keywords>
  <TimeCreated SystemTime="2023-06-05T18:50:35.245831000Z"/>
  <EventRecordID>934</EventRecordID>
  <Correlation/>
  <Execution ProcessID="3948" ThreadID="3948"/>
  <Channel>Linux-Sysmon/Operational</Channel>
  <Computer>202-u20-dev-agent</Computer>
  <Security UserId="0"/>
</System>
<EventData>
  <Data Name="RuleName">-</Data>
  <Data Name="UtcTime">2023-06-05 18:50:35.246</Data>
  <Data Name="ProcessGuid">{924935dc-2e7b-647e-f644-17b56f550000}</Data>
  <Data Name="ProcessId">4276</Data>
  <Data Name="Image">/usr/bin/touch</Data>
  <Data Name="FileVersion">-</Data>
  <Data Name="Description">-</Data>
  <Data Name="Product">-</Data>
  <Data Name="Company">-</Data>
  <Data Name="OriginalFileName">-</Data>
  <Data Name="CommandLine">touch /tmp/e11.log</Data>
  <Data Name="CurrentDirectory">/root</Data>
  <Data Name="User">root</Data>
  <Data Name="LogonGuid">{924935dc-fbf9-647d-0000-000001000000}</Data>
  <Data Name="LogonId">0</Data>
  <Data Name="TerminalSessionId">7</Data>
  <Data Name="IntegrityLevel">no level</Data>
  <Data Name="Hashes">-</Data>
  <Data Name="ParentProcessGuid">{924935dc-09fc-647e-0d6f-e4e7d3550000}</Data>
  <Data Name="ParentProcessId">3426</Data>
  <Data Name="ParentImage">/usr/bin/bash</Data>
  <Data Name="ParentCommandLine">bash</Data>
  <Data Name="ParentUser">root</Data>
</EventData>
</Event>
Jun  5 18:50:35 202-u20-dev-agent sysmon: <Event>
<System>
<Provider Name="Linux-Sysmon" Guid="{ff032593-a8d3-4f13-b0d6-01fc615a0f97}"/>
<EventID>11</EventID>
<Version>2</Version>
<Level>4</Level>
<Task>11</Task>
<Opcode>0</Opcode>
<Keywords>0x8000000000000000</Keywords>
<TimeCreated SystemTime="2023-06-05T18:50:35.246709000Z"/>
<EventRecordID>935</EventRecordID>
<Correlation/>
<Execution ProcessID="3948" ThreadID="3948"/>
<Channel>Linux-Sysmon/Operational</Channel>
<Computer>202-u20-dev-agent</Computer>
<Security UserId="0"/>
</System>
<EventData>
<Data Name="RuleName">-</Data>
<Data Name="UtcTime">2023-06-05 18:50:35.248</Data>
<Data Name="ProcessGuid">{924935dc-2e7b-647e-f644-17b56f550000}</Data>
<Data Name="ProcessId">4276</Data>
<Data Name="Image">/usr/bin/touch</Data>
<Data Name="TargetFilename">/tmp/e11.log</Data>
<Data Name="CreationUtcTime">2023-06-05 18:50:35.248</Data>
<Data Name="User">-</Data>
</EventData>
</Event>

<!--
  Event id 1 ==  ProcessCreate
  Event id 13 ==  FileDelete
  rm /tmp/e11.log
-->
Jun  5 18:51:25 202-u20-dev-agent sysmon:<Event>
<System>
<Provider Name="Linux-Sysmon" Guid="{ff032593-a8d3-4f13-b0d6-01fc615a0f97}"/>
<EventID>1</EventID>
<Version>5</Version>
<Level>4</Level>
<Task>1</Task>
<Opcode>0</Opcode>
<Keywords>0x8000000000000000</Keywords>
<TimeCreated SystemTime="2023-06-05T18:51:25.727868000Z"/>
<EventRecordID>936</EventRecordID>
<Correlation/>
<Execution ProcessID="3948" ThreadID="3948"/>
<Channel>Linux-Sysmon/Operational</Channel>
<Computer>202-u20-dev-agent</Computer>
<Security UserId="0"/>
</System>
<EventData>
<Data Name="RuleName">-</Data>
<Data Name="UtcTime">2023-06-05 18:51:25.728</Data>
<Data Name="ProcessGuid">{924935dc-2ead-647e-56e5-ef79de550000}</Data>
<Data Name="ProcessId">4279</Data>
<Data Name="Image">/usr/bin/rm</Data>
<Data Name="FileVersion">-</Data>
<Data Name="Description">-</Data>
<Data Name="Product">-</Data>
<Data Name="Company">-</Data>
<Data Name="OriginalFileName">-</Data>
<Data Name="CommandLine">rm /tmp/e11.log</Data>
<Data Name="CurrentDirectory">/root</Data>
<Data Name="User">root</Data>
<Data Name="LogonGuid">{924935dc-fbf9-647d-0000-000001000000}</Data>
<Data Name="LogonId">0</Data>
<Data Name="TerminalSessionId">7</Data>
<Data Name="IntegrityLevel">no level</Data>
<Data Name="Hashes">-</Data>
<Data Name="ParentProcessGuid">{924935dc-09fc-647e-0d6f-e4e7d3550000}</Data>
<Data Name="ParentProcessId">3426</Data>
<Data Name="ParentImage">/usr/bin/bash</Data>
<Data Name="ParentCommandLine">bash</Data>
<Data Name="ParentUser">root</Data>
</EventData>
</Event>
Jun  5 18:51:25 202-u20-dev-agent sysmon: <Event>
<System>
<Provider Name="Linux-Sysmon" Guid="{ff032593-a8d3-4f13-b0d6-01fc615a0f97}"/>
<EventID>23</EventID>
<Version>5</Version>
<Level>4</Level>
<Task>23</Task>
<Opcode>0</Opcode>
<Keywords>0x8000000000000000</Keywords>
<TimeCreated SystemTime="2023-06-05T18:51:25.730031000Z"/>
<EventRecordID>937</EventRecordID>
<Correlation/>
<Execution ProcessID="3948" ThreadID="3948"/>
<Channel>Linux-Sysmon/Operational</Channel>
<Computer>202-u20-dev-agent</Computer>
<Security UserId="0"/>
</System>
<EventData>
<Data Name="RuleName">-</Data>
<Data Name="UtcTime">2023-06-05 18:51:25.731</Data>
<Data Name="ProcessGuid">{924935dc-2ead-647e-56e5-ef79de550000}</Data>
<Data Name="ProcessId">4279</Data>
<Data Name="User">-</Data>
<Data Name="Image">/usr/bin/rm</Data>
<Data Name="TargetFilename">/tmp/e11.log</Data>
<Data Name="Hashes">-</Data>
<Data Name="IsExecutable">-</Data>
<Data Name="Archived">-</Data>
</EventData>
</Event>

<!--
  Event id 3 ==  NetworkConnect
  ping google.com
-->
Jun  5 18:55:32 202-u20-dev-agent sysmon:<Event>
<System>
<Provider Name="Linux-Sysmon" Guid="{ff032593-a8d3-4f13-b0d6-01fc615a0f97}"/>
<EventID>3</EventID>
<Version>5</Version>
<Level>4</Level>
<Task>3</Task>
<Opcode>0</Opcode>
<Keywords>0x8000000000000000</Keywords>
<TimeCreated SystemTime="2023-06-05T18:55:32.369089000Z"/>
<EventRecordID>953</EventRecordID>
<Correlation/>
<Execution ProcessID="4365" ThreadID="4365"/>
<Channel>Linux-Sysmon/Operational</Channel>
<Computer>202-u20-dev-agent</Computer>
<Security UserId="0"/>
</System>
<EventData>
<Data Name="RuleName">-</Data>
<Data Name="UtcTime">2023-06-05 18:55:32.374</Data>
<Data Name="ProcessGuid">{924935dc-2fa4-647e-4631-a2664d560000}</Data>
<Data Name="ProcessId">4367</Data>
<Data Name="Image">/usr/bin/ping</Data>
<Data Name="User">root</Data>
<Data Name="Protocol">udp</Data>
<Data Name="Initiated">false</Data>
<Data Name="SourceIsIpv6">false</Data>
<Data Name="SourceIp">127.0.0.53</Data>
<Data Name="SourceHostname">-</Data>
<Data Name="SourcePort">53</Data>
<Data Name="SourcePortName">-</Data>
<Data Name="DestinationIsIpv6">false</Data>
<Data Name="DestinationIp">127.0.0.1</Data>
<Data Name="DestinationHostname">-</Data>
<Data Name="DestinationPort">60125</Data>
<Data Name="DestinationPortName">-</Data>
</EventData>
</Event>


<!--
  Event id 9 == RawAccessRead
  dd if=/dev/sda bs=512 count=1 1>/dev/null
-->
Jun  5 18:59:01 202-u20-dev-agent sysmon:<Event>
<System>
<Provider Name="Linux-Sysmon" Guid="{ff032593-a8d3-4f13-b0d6-01fc615a0f97}"/>
<EventID>9</EventID>
<Version>2</Version>
<Level>4</Level>
<Task>9</Task>
<Opcode>0</Opcode>
<Keywords>0x8000000000000000</Keywords>
<TimeCreated SystemTime="2023-06-05T18:59:01.502659000Z"/>
<EventRecordID>1071</EventRecordID>
<Correlation/>
<Execution ProcessID="4365" ThreadID="4365"/>
<Channel>Linux-Sysmon/Operational</Channel>
<Computer>202-u20-dev-agent</Computer>
<Security UserId="0"/>
</System>
<EventData>
<Data Name="RuleName">-</Data>
<Data Name="UtcTime">2023-06-05 18:59:01.508</Data>
<Data Name="ProcessGuid">{924935dc-3075-647e-36cc-43da69550000}</Data>
<Data Name="ProcessId">4466</Data>
<Data Name="Image">/usr/bin/dd</Data>
<Data Name="Device">/dev/sda</Data>
<Data Name="User">-</Data>
</EventData>
</Event>






```

### CODE REVIEW

File: src/engine/ruleset/integrations/system/tests/sysmon-linux/eventID_5_expected.json [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+            "action": "process_terminated",
+            "outcome": "success",
+            "provider": "Linux-Sysmon"
+        },
+        "host": {
+            "id": "002",
+            "hostname": "202-u20-dev-agent",
+            "name": "202-u20-dev-agent"
+        },
+        "~TAG": "sysmon",
```
Comment (LucioDonda) [inline src/engine/ruleset/integrations/system/tests/sysmon-linux/eventID_5_expected.json:107]: should this be present on the final event?

**Review by LucioDonda — APPROVED** [review]
LGTM!

**Review by JavierBejMen — APPROVED** [review]
LGTM

**Review by nbertoldo — APPROVED** [review]
LGTM!

**Review by JcabreraC — APPROVED** [review]
LGTM !

**Pull request merged**